packages/gatsby/src/utils/parcel/compile-gatsby-files.ts

```diff
@@ -100,8 +102,22 @@ export async function compileGatsbyFiles(
 
     await exponentialBackoff(retry)
 
-    const parcel = constructParcel(siteRoot)
+    // for whatever reason TS thinks LMDBCache is some browser Cache and not actually Parcel's Cache
+    // so we force type it to Parcel's Cache
+    const cache = new LMDBCache(getCacheDir(siteRoot)) as any as Cache
```

Comment:
```suggestion
    const cache = new LMDBCache(getCacheDir(siteRoot)) as unknown as Cache
```

Normally it's like that. Does that work, too?